pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,945 @@
+
+        .. note::
+
+            Ensure that no power is applied to the sensor while the power meter
+            is zeroing. Any applied RF input power will cause an erroneous
+            reading.
+
+        """
+        self.write("ZE")
+
+    trigger_mode = Instrument.control(
+        "SM", "TR%d",
+        """
+        Control the trigger mode.
+
+        The power meter has two modes of triggered operation; standby mode and free run mode.
+        Standby mode means the power meter is making measurements, but the display and HP-IB are
+        not updated until a trigger command is received. Free run means that Meter takes
+        measurements and updates the display and HP-IB continuously.
+        """,
+        values=[e for e in TriggerMode],
+        validator=strict_discrete_set,
+        get_process=__getstatus(StatusMessage.TriggerMode, lambda v: TriggerMode(v)),
+        set_process=lambda v: int(v)
+    )
+
+    def trigger_immediate(self):
+        """
+        Trigger immediate.
+
+        When the power meter receives the trigger immediate program code, it inputs one more data
+        point into the digital filter, measures the reading from the filter, and then updates
+        the display and HP-IB. (When the trigger immediate command is executed, the internal
+        digital filter is not cleared.) The power meter then waits for the measurement results to
+        be read by the controller. While waiting, the power meter can process most bus commands
+        without losing the measurement results. If the power meter receives a trigger immediate
+        command and then receives the GET (Group Execute Trigger) command, the trigger immediate
+        command will be aborted and a new measurement cycle will be executed. Once the
+        measurement results are read onto the bus, the power meter always reverts to standby/hold
+        mode. Measurement results obtained via trigger immediate are normally valid only when the
+        power meter is in a steady, settled state.
+        """
+        self.write("TR1")
+
+    def trigger_delay(self):
+        """
+        Trigger with delay.
+
+        Triggering with delay is identical to :meth:`trigger_immediate` except the power meter
+        inserts a settling-time delay before taking the requested measurement.
+        This settling time allows the internal digital filter to be updated with new values to
+        produce valid, accurate measurement results. The trigger with delay command allows time
+        for settling of the internal amplifiers and filters. It does not allow time for power
+        sensor delay. In cases of large power changes, the delay may not be sufficient for
+        complete settling. Accurate readings can be assured by taking two successive measurements
+        for comparison. Once the measurement results are displayed and read onto the bus,
+        the power meter reverts to standby mode.
+        """
+        self.trigger_delay()
```

Comment:
That is an infinite recursion, as it calls itself again!.

Reply:
Yeah thanks, that was a mistake ...